orttraining/orttraining/python/training/_ortmodule_io.py

```diff
@@ -1,19 +1,50 @@
 from collections import abc
 import copy
-import functools
 import inspect
 import torch
 import warnings
 
 
 class _InputInfo(object):
-    def __init__(self, names, shape, require_grad_names=None, dynamic_axes=None, schema=None):
+    def __init__(self,
+                 names,
+                 shape,
+                 require_grad_names=None,
+                 dynamic_axes=None,
+                 schema=None,
+                 num_positionals=0,
+                 keyword_names=None):
         self.names = names
         self.shape = shape
         self.require_grad_names = require_grad_names if require_grad_names else []
         self.dynamic_axes = dynamic_axes if dynamic_axes else {}
         self.schema = schema if schema else []
+        self.num_positionals = num_positionals
+        self.keyword_names = keyword_names
 
+    def __repr__(self) -> str:
+        return f'''_InputInfo class:
+            \tNames:            {self.names}
+            \tShape:            {self.shape}
+            \tRequire gradient: {self.require_grad_names}
+            \tDynamic axes:     {self.dynamic_axes}
+            \tSchema:           {self.schema}
+            \t#Positionals:     {self.num_positionals}
+            \tKeyword names:    {self.keyword_names}'''
+
+    def flatten(self, args, kwargs):
+        assert len(args) == self.num_positionals
+        assert set(kwargs.keys()) == set(self.keyword_names)
```

Comment:
Although I like how these assertions are self-documenting, they might be a little expensive (especially assuming we're checking this elsewhere), and we should probably not evaluate them at runtime.